operator/src/lib.rs

```diff
@@ -0,0 +1,608 @@
+    }
+
+    fn build_pod(
+        &self,
+        selector: &SparkNodeSelector,
+        hash: &str,
+        node_type: &SparkNodeType,
+    ) -> Result<Pod, Error> {
+        Ok(Pod {
+            metadata: self.build_pod_metadata(node_type, hash)?,
+            spec: Some(self.build_pod_spec(selector, node_type, hash)),
+            ..Pod::default()
+        })
+    }
+
+    fn build_pod_metadata(
+        &self,
+        node_type: &SparkNodeType,
+        hash: &str,
+    ) -> Result<ObjectMeta, Error> {
+        Ok(ObjectMeta {
+            labels: Some(self.build_labels(node_type, hash)),
+            name: Some(self.create_pod_name(node_type, hash)),
+            namespace: Some(self.context.namespace()),
+            owner_references: Some(vec![metadata::object_to_owner_reference::<SparkCluster>(
+                self.context.metadata(),
+            )?]),
+            ..ObjectMeta::default()
+        })
+    }
+
+    fn build_pod_spec(
+        &self,
+        selector: &SparkNodeSelector,
+        node_type: &SparkNodeType,
+        hash: &str,
+    ) -> PodSpec {
+        let (containers, volumes) = self.build_containers(node_type, hash);
+
+        PodSpec {
+            node_name: Some(selector.node_name.clone()),
+            tolerations: Some(create_tolerations()),
+            containers,
+            volumes: Some(volumes),
+            ..PodSpec::default()
+        }
+    }
+
+    fn build_containers(
+        &self,
+        node_type: &SparkNodeType,
+        hash: &str,
+    ) -> (Vec<Container>, Vec<Volume>) {
+        // TODO: get version from controller
+        let image_name = "spark:3.0.1".to_string();
+
+        // adapt worker command with master url(s)
+        let mut command = vec![node_type.get_command()];
+        let adapted_command = config::adapt_container_command(node_type, &self.spec.master);
+        if !adapted_command.is_empty() {
+            command.push(adapted_command);
+        }
+
+        let containers = vec![Container {
+            image: Some(image_name),
+            name: "spark".to_string(),
+            // TODO: worker -> add master port
+            command: Some(command),
+            volume_mounts: Some(vec![
+                // One mount for the config directory, this will be relative to the extracted package
+                VolumeMount {
+                    mount_path: "conf".to_string(),
+                    name: "config-volume".to_string(),
+                    ..VolumeMount::default()
+                },
+                VolumeMount {
+                    mount_path: "/tmp".to_string(), // TODO: Make configurable via crd log dir
+                    name: "event-volume".to_string(),
+                    ..VolumeMount::default()
+                },
+            ]),
+            // set no daemonize and configroot via env variables:
+            // if loaded via spark-default.conf or spark-env.sh the variables are set too late!
```

Comment:
Could you document why we need this at all? 
The NO_DAEMONIZE.
Just a sentence that otherwise systemdo won't work or whatever the reason is.